### PR TITLE
WIP Blob/Object get

### PR DESCRIPTION
**Pull request by sydney-munro**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-storage/src/test/java/com/google/cloud/storage/jqwik/StorageArbitraries.java [hunk 1]
```diff
@@ -137,4 +151,48 @@ public String get() {
...
+
+    public Arbitrary<ObjectChecksums> objectChecksumsArbitrary() {
+      return Combinators.combine(
+              Arbitraries.integers().greaterOrEqual(1),
+              Arbitraries.strings()
+                  .map(
+                      s ->
+                          BaseEncoding.base64()
+                              .encode(Hashing.md5().hashBytes(s.getBytes()).asBytes())))
+          .as(
```
Comment (BenWhitehead) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/jqwik/StorageArbitraries.java:177]: This can be done in a later PR if you feel like it.

This will need to changes from a combine of two arbitraries, to a single arbitrary which yields both crc32c and md5 values.

We want to ensure the value for those fields is generated base on the same bytes in the general case. Later we can investigate if we want to add an edge case where the md5 does not correspond to crc32c of the same input data.

File: google-cloud-storage/src/test/java/com/google/cloud/storage/jqwik/StorageArbitraries.java [hunk 2]
```diff
@@ -137,4 +151,48 @@ public String get() {
...
+              Arbitraries.strings()
+                  .map(s -> Hashing.sha256().hashString(s, StandardCharsets.UTF_8).asBytes()))
+          .as(
+              (algorithm, key) ->
+                  CustomerEncryption.newBuilder()
+                      .setEncryptionAlgorithm(algorithm)
+                      .setKeySha256Bytes(ByteString.copyFrom(key))
+                      .build());
+    }
+  }
```
Comment (BenWhitehead) [inline google-cloud-storage/src/test/java/com/google/cloud/storage/jqwik/StorageArbitraries.java:197]: Nit: move the ByteString copy up in the arbitrary generation
```suggestion
    public Arbitrary<CustomerEncryption> customerEncryptionArbitrary() {
      return Combinators.combine(
              Arbitraries.strings().ofMinLength(1).ofMaxLength(1024),
              Arbitraries.strings()
                  .map(s -> Hashing.sha256().hashString(s, StandardCharsets.UTF_8).asBytes())
                  .map(ByteString::copyFrom))
          .as(
              (algorithm, key) ->
                  CustomerEncryption.newBuilder()
                      .setEncryptionAlgorithm(algorithm)
                      .setKeySha256Bytes(key)
                      .build());
    }

```

**Review by BenWhitehead — CHANGES_REQUESTED** [review]
Overall looks pretty good. A few changes about base64 handling need to be addressed, but other than that looks pretty good.

File: google-cloud-storage/src/test/java/com/google/cloud/storage/jqwik/StorageArbitraries.java (hunk 1 above)
Comment (sydney-munro) [reply google-cloud-storage/src/test/java/com/google/cloud/storage/jqwik/StorageArbitraries.java:177]: ah I see makes sense. I will look into it.

**Review by BenWhitehead — APPROVED** [review]
